src/zkscript/transaction_introspection/transaction_introspection.py

```diff
@@ -0,0 +1,220 @@
+        # stack out: [GROUP_ORDER_INT, Gx, Gx_bytes, .., sig_hash_preimage, ..,
+        #               Gx_bytes, 0x0220||Gx_bytes||02, h]
+        out += move(sig_hash_preimage.shift(2), bool_to_moving_function(rolling_option))
+        out += Script.parse_string("OP_HASH256") + reverse_endianness(32) + bytes_to_unsigned()
+
+        # Compute the s part of the signature
+        # stack in:  [GROUP_ORDER_INT, Gx, Gx_bytes, .., sig_hash_preimage, ..,
+        #               Gx_bytes, 0x0220||Gx_bytes||02, h]
+        # stack out: [GROUP_ORDER_INT, Gx, Gx_bytes, .., sig_hash_preimage, ..,
+        #               Gx_bytes, 0x0220||Gx_bytes||02, GROUP_ORDER_INT, (h + Gx) % GROUP_ORDER_INT]
+        out += roll(position=-2, n_elements=1) if clean_constants else pick(position=-2, n_elements=1)  # Move Gx
+        out += Script.parse_string("OP_ADD")
+        out += (
+            roll(position=-1, n_elements=1) if clean_constants else pick(position=-1, n_elements=1)
+        )  # Move GROUP_ORDER_INT
+        out += Script.parse_string("OP_TUCK OP_MOD")
+
+        # stack in:  [GROUP_ORDER_INT, Gx, Gx_bytes, .., sig_hash_preimage, ..,
+        #               Gx_bytes, 0x0220||Gx_bytes||02 GROUP_ORDER_INT, (h + Gx) % GROUP_ORDER_INT]
+        # stack out: [GROUP_ORDER_INT, Gx, Gx_bytes, .., sig_hash_preimage, ..,
+        #               Gx_bytes, 0x0220||Gx_bytes||02, s]
+        out += int_sig_to_s_component(add_prefix=False)
+
+        # stack in: [GROUP_ORDER_INT, Gx, Gx_bytes, .., sig_hash_preimage, ..,
+        #               Gx_bytes, 0x0220||Gx_bytes||02, s]
+        # stack out: [GROUP_ORDER_INT, Gx, Gx_bytes, .., sig_hash_preimage, ..,
+        #               Gx_bytes, Der(Gx,s)]
+        out += Script.parse_string(
+            "OP_SIZE OP_TUCK OP_TOALTSTACK OP_CAT OP_CAT"
+        )  # Construct 0x0220||Gx||02||len(s)||s and put len(s) on the altstack
+        out += Script.parse_string("0x30 OP_FROMALTSTACK")
+        out += nums_to_script([36])
+        out += Script.parse_string("OP_ADD OP_CAT OP_SWAP OP_CAT")  # Construct DER(Gx,s)
+        out.append_pushdata(sighash_value.to_bytes())
+        out += Script.parse_string("OP_CAT")  # Append SIGHASH
+
+        # stack in:  [GROUP_ORDER_INT, Gx, Gx_bytes, .., sig_hash_preimage, ..,
+        #               Gx_bytes, Der(Gx,s)]
+        # stack out: [GROUP_ORDER_INT, Gx, Gx_bytes, .., sig_hash_preimage, ..,
+        #               Der(Gx,s), Gx_bytes]
+        out += Script.parse_string("OP_2 OP_ROT OP_CAT")
+
+        if is_checksigverify and not is_opcodeseparator:
+            out += Script.parse_string("OP_CHECKSIGVERIFY")
+        elif is_checksigverify and is_opcodeseparator:
+            out += Script.parse_string("OP_CODESEPARATOR OP_CHECKSIGVERIFY")
+        elif not is_checksigverify and is_opcodeseparator:
+            out += Script.parse_string("OP_CODESEPARATOR OP_CHECKSIG")
+        else:
+            out += Script.parse_string("OP_CHECKSIG")
+
```

Comment:
if is_opcodeseparator:
    out += Script.parse_string("OP_CODESEPARATOR")
out += Script.parse_string("OP_CHECKSIGVERIFY" if is_checksigverify else "OP_CHECKSIG")

Reply:
not a big change, but should make it a bit more readable